src/server/acl/acl_family.cc

```diff
@@ -84,146 +47,38 @@ void AclFamily::List(CmdArgList args, ConnectionContext* cntx) {
-      return {};
-    }
-    return {res->second, true};
-  }
-
-  if (absl::StartsWith(command, "-@")) {
-    auto res = CATEGORY_INDEX_TABLE.find(command.substr(2));
-    if (res == CATEGORY_INDEX_TABLE.end()) {
-      return {};
-    }
-    return {res->second, false};
-  }
-
-  return {};
-}
-
-using facade::ErrorReply;
-
-template <typename T>
-std::variant<User::UpdateRequest, ErrorReply> ParseAclSetUser(T args, bool hashed = false) {
-  User::UpdateRequest req;
-
-  for (auto& arg : args) {
-    if (auto pass = MaybeParsePassword(facade::ToSV(arg)); pass) {
-      if (req.password) {
-        return ErrorReply("Only one password is allowed");
-      }
-      req.password = std::move(pass);
-      req.is_hashed = hashed;
-      continue;
-    }
-    if constexpr (std::is_same_v<T, CmdArgList>) {
-      ToUpper(&arg);
-    }
-    const auto command = facade::ToSV(arg);
-
-    if (auto status = MaybeParseStatus(command); status) {
-      if (req.is_active) {
-        return ErrorReply("Multiple ON/OFF are not allowed");
-      }
-      req.is_active = *status;
-      continue;
-    }
-
-    auto [cat, add] = MaybeParseAclCategory(command);
-    if (!cat) {
-      return ErrorReply(absl::StrCat("Unrecognized parameter ", command));
-    }
-
-    using Sign = User::Sign;
-    using Val = std::pair<Sign, uint32_t>;
-    auto val = add ? Val{Sign::PLUS, *cat} : Val{Sign::MINUS, *cat};
-    req.categories.push_back(val);
-  }
-
-  return req;
-}
-
-using MaterializedContents = std::optional<std::vector<std::vector<std::string_view>>>;
-
-MaterializedContents MaterializeFileContents(std::vector<std::string>* usernames,
-                                             std::string_view file_contents) {
-  // This is fine, a very large file will top at 1-2 mb. And that's for 5000+ users with 400
-  // characters of ACL infor...
-  std::vector<std::string_view> commands = absl::StrSplit(file_contents, "\n");
-  std::vector<std::vector<std::string_view>> materialized;
-  materialized.reserve(commands.size());
-  usernames->reserve(commands.size());
-  for (auto& command : commands) {
-    if (command.empty())
-      continue;
-    std::vector<std::string_view> cmds = absl::StrSplit(command, ' ');
-    if (cmds[0] != "ACL" || cmds[1] != "SETUSER" || cmds.size() < 3) {
-      return {};
-    }
-    usernames->push_back(std::string(cmds[2]));
-    cmds.erase(cmds.begin(), cmds.begin() + 3);
-    materialized.push_back(cmds);
-  }
-  return materialized;
-}
-
-}  // namespace
-
 void AclFamily::StreamUpdatesToAllProactorConnections(const std::vector<std::string>& user,
-                                                      const std::vector<uint32_t>& update_cat) {
-  auto update_cb = [&user, &update_cat]([[maybe_unused]] size_t id, util::Connection* conn) {
+                                                      const std::vector<uint32_t>& update_cat,
+                                                      const NestedVector& update_commands) {
```

Comment:
@dranikpg Two things here:

1. The inner vector of `NestedVector` is relatively small. Remember that, it's a vector of size 15 or 14 (depending if we are running on mac where search is not enabled there).
2. I remember you saying that we could use a `shared_ptr` here. I gave it some thought and I would argue against it because copying a few kb's of vectors is insignificant with the cost of accessing memory on different NUMA cores, since the accesses are proportional to the distance of the pinned memory chip. Copying is just fine here.
3. I am planning to measure this at some point just to be on the safe side.

Reply:
No no, I mean the actual access of the memory contents is going to be slower, because other cores will need to access memory that is pinned on some other core.

Reply:
An interesting talk about numa in general https://www.youtube.com/watch?v=f0ZKBusa4CI